src/QAT/qat/purr/compiler/devices.py

```diff
@@ -446,6 +447,24 @@ def __hash__(self):
         return hash(self.full_id())
 
 
+class FreqShiftPulseChannel(PulseChannel):
+    def __init__(
+        self,
+        id_: str,
+        physical_channel: PhysicalChannel,
+        frequency=0.0,
+        bias=0.0 + 0.0j,
+        scale=1.0 + 0.0j,
+        amp=0.0,
+        active: bool = True,
+        fixed_if: bool = False,
+        **kwargs
```

Comment:
Better to just define the rest of the args here since you're a child (and already have for most of them)